src/internal/observable/zip.ts

```diff
@@ -88,7 +88,7 @@ export function zip(...args: unknown[]): Observable<unknown> {
                   // If any one of the sources is both complete and has an empty buffer
                   // then we complete the result. This is because we cannot possibly have
                   // any more values to zip together.
-                  if (buffers.some((buffer, i) => !buffer.length && completed[i])) {
+                  if (buffers !== null && buffers.some((buffer, i) => !buffer.length && completed[i])) {
```

Comment:
nit:

```suggestion
                  if (buffers?.some((buffer, i) => !buffer.length && completed[i])) {
```

Reply:
I initially thought using an explicit `!== null` check is better as it results in a boolean which some people (including myself :-)) prefer. But I think your suggestion is more in line with the existing RxJS codebase.

I've changed it. Thanks for the review.